proof/refine/RISCV64/KHeap_R.thy

```diff
@@ -2943,6 +2996,10 @@ lemma ep_redux_simps':
   by (fastforce split: list.splits
                 simp: valid_ep_def valid_ntfn_def)+
 
+lemma endpoint_live':
+  "\<lbrakk>ko_at' ep ptr s; ep \<noteq> IdleEP\<rbrakk> \<Longrightarrow> ko_wp_at' live' ptr s"
+  apply (clarsimp simp: ko_wp_at'_def obj_at'_def)
+  done
```

Comment:
by?

Reply:
I generally prefer using `by` only when the last step or two takes more than a few seconds. I sometimes use it for proofs which are very short and which we wouldn't expect to change often. Is there enough of a preference for `by` for me to change these? Or to go through and change a whole bunch of them?

Reply:
From my perspective: It doesn't look like an interesting proof, or one that's likely to break. The extra line wastes visual space, adding nothing. If it breaks, you'll need to change something on the one line anyway, at which point having to change by to apply won't matter much. @lsf37 might have a different view, but I picked up the habit from him

Reply:
I tend to use `by` for most one-liners, yes. I don't have an extremely strong opinion on it, but I'd say a preference for.